// @neynar/nodejs-sdk/build/api/apis/studio-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Neynar API
 * The Neynar API allows you to interact with the Farcaster protocol among other things. See the [Neynar docs](https://docs.neynar.com/reference) for more details.
 *
 * The version of the OpenAPI document: 3.137.0
 * Contact: team@neynar.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioApi = exports.StudioApiFactory = exports.StudioApiFp = exports.StudioApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * StudioApi - axios parameter creator
 * @export
 */
const StudioApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Associates a generated miniapp with a Farcaster account using a JFS-signed domain association. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Set account association
         * @param {AssociateDeploymentRequest} associateDeploymentRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<AssociateDeployment200Response>} A promise that resolves to a `AssociateDeployment200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/associate-deployment)
         *
         */
        associateDeployment: async (associateDeploymentRequest, options = {}) => {
            // verify required parameter 'associateDeploymentRequest' is not null or undefined
            (0, common_1.assertParamExists)('associateDeployment', 'associateDeploymentRequest', associateDeploymentRequest);
            const localVarPath = `/v2/studio/deployment/account-association`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(associateDeploymentRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Runs Next.js build process for the generated app. If build fails, automatically calls a build-fixer agent to resolve errors. Streams build output and agent responses via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Build generated app with automatic error fixing
         * @param {BuildRequest} buildRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/build)
         *
         */
        build: async (buildRequest, options = {}) => {
            // verify required parameter 'buildRequest' is not null or undefined
            (0, common_1.assertParamExists)('build', 'buildRequest', buildRequest);
            const localVarPath = `/v2/studio/deployment/build`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(buildRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Creates and deploys an instance of the miniapp generator for a user. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Create a miniapp generator deployment
         * @param {CreateDeploymentRequest} createDeploymentRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/create-deployment)
         *
         */
        createDeployment: async (createDeploymentRequest, options = {}) => {
            // verify required parameter 'createDeploymentRequest' is not null or undefined
            (0, common_1.assertParamExists)('createDeployment', 'createDeploymentRequest', createDeploymentRequest);
            const localVarPath = `/v2/studio/deployment/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createDeploymentRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes a specific miniapp generator deployment or all deployments for a FID. If deployment_id or name is provided, deletes single deployment. If only FID is provided, deletes all deployments for that FID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Delete deployment(s)
         * @param {DeleteDeploymentRequest} deleteDeploymentRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeleteDeployment200Response>} A promise that resolves to a `DeleteDeployment200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-deployment)
         *
         */
        deleteDeployment: async (deleteDeploymentRequest, options = {}) => {
            // verify required parameter 'deleteDeploymentRequest' is not null or undefined
            (0, common_1.assertParamExists)('deleteDeployment', 'deleteDeploymentRequest', deleteDeploymentRequest);
            const localVarPath = `/v2/studio/deployment/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(deleteDeploymentRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk deletes.
         * @summary Delete rows from table
         * @param {string} tableName
         * @param {DeleteRowsRequest} deleteRowsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeleteRows200Response>} A promise that resolves to a `DeleteRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-rows)
         *
         */
        deleteRows: async (tableName, deleteRowsRequest, options = {}) => {
            // verify required parameter 'tableName' is not null or undefined
            (0, common_1.assertParamExists)('deleteRows', 'tableName', tableName);
            // verify required parameter 'deleteRowsRequest' is not null or undefined
            (0, common_1.assertParamExists)('deleteRows', 'deleteRowsRequest', deleteRowsRequest);
            const localVarPath = `/v2/studio/deployment/database/tables/{table_name}/rows`
                .replace(`{${"table_name"}}`, encodeURIComponent(String(tableName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(deleteRowsRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes environment variables (secrets) from a deployment.
         * @summary Delete deployment secrets
         * @param {DeleteSecretsRequest} deleteSecretsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-secrets)
         *
         */
        deleteSecrets: async (deleteSecretsRequest, options = {}) => {
            // verify required parameter 'deleteSecretsRequest' is not null or undefined
            (0, common_1.assertParamExists)('deleteSecrets', 'deleteSecretsRequest', deleteSecretsRequest);
            const localVarPath = `/v2/studio/deployment/secrets/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(deleteSecretsRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Publishes the generated miniapp to Vercel via GitHub. Creates a GitHub repository, pushes code, creates a Vercel project linked to GitHub, and triggers deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Deploy miniapp to Vercel
         * @param {DeployToVercelRequest} deployToVercelRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeployToVercel200Response>} A promise that resolves to a `DeployToVercel200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/deploy-to-vercel)
         *
         */
        deployToVercel: async (deployToVercelRequest, options = {}) => {
            // verify required parameter 'deployToVercelRequest' is not null or undefined
            (0, common_1.assertParamExists)('deployToVercel', 'deployToVercelRequest', deployToVercelRequest);
            const localVarPath = `/v2/studio/vercel/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(deployToVercelRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Executes a raw SQL query against the deployment database. Only SELECT, WITH, and EXPLAIN queries are allowed. Admin access required.
         * @summary Execute raw SQL query (admin only)
         * @param {ExecuteSqlRequest} executeSqlRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ExecuteSql200Response>} A promise that resolves to a `ExecuteSql200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/execute-sql)
         *
         */
        executeSql: async (executeSqlRequest, options = {}) => {
            // verify required parameter 'executeSqlRequest' is not null or undefined
            (0, common_1.assertParamExists)('executeSql', 'executeSqlRequest', executeSqlRequest);
            const localVarPath = `/v2/studio/deployment/database/sql`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(executeSqlRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves the account-association.json file from a miniapp deployment, which contains the JFS-signed domain association. Requires API key authentication.
         * @summary Get account association of a miniapp
         * @param {string} [deploymentId] Deployment ID
         * @param {string} [namespace] Kubernetes namespace name
         * @param {string} [name] Kubernetes deployment name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetAccountAssociation200Response>} A promise that resolves to a `GetAccountAssociation200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-account-association)
         *
         */
        getAccountAssociation: async (deploymentId, namespace, name, options = {}) => {
            const localVarPath = `/v2/studio/deployment/account-association`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves all messages in a specific conversation. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get messages in a conversation
         * @param {string} conversationId Conversation ID
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {boolean | null} [includeDeleted] Include deleted messages in the response. Defaults to false.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetConversationMessages200Response>} A promise that resolves to a `GetConversationMessages200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-conversation-messages)
         *
         */
        getConversationMessages: async (conversationId, deploymentId, fid, name, namespace, includeDeleted, options = {}) => {
            // verify required parameter 'conversationId' is not null or undefined
            (0, common_1.assertParamExists)('getConversationMessages', 'conversationId', conversationId);
            const localVarPath = `/v2/studio/deployment/conversations/messages`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (conversationId !== undefined) {
                localVarQueryParameter['conversation_id'] = conversationId;
            }
            if (includeDeleted !== undefined) {
                localVarQueryParameter['include_deleted'] = includeDeleted;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetches info about a miniapp generator deployment by its deployment_id or name and creator\'s Farcaster ID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get deployment info
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment)
         *
         */
        getDeployment: async (deploymentId, fid, name, namespace, options = {}) => {
            const localVarPath = `/v2/studio/deployment/by-name-and-fid`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves the contents of a specific file from the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get deployment file contents
         * @param {string} filePath File path relative to gen/
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetDeploymentFile200Response>} A promise that resolves to a `GetDeploymentFile200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment-file)
         *
         */
        getDeploymentFile: async (filePath, deploymentId, fid, name, namespace, options = {}) => {
            // verify required parameter 'filePath' is not null or undefined
            (0, common_1.assertParamExists)('getDeploymentFile', 'filePath', filePath);
            const localVarPath = `/v2/studio/deployment/file`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (filePath !== undefined) {
                localVarQueryParameter['file_path'] = filePath;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves the dev-status.json file from a miniapp deployment, which tracks the progress of app development phases. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get dev status of a miniapp
         * @param {string} [deploymentId] Deployment ID
         * @param {string} [namespace] Kubernetes namespace name
         * @param {string} [name] Kubernetes deployment name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetDevStatus200Response>} A promise that resolves to a `GetDevStatus200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-dev-status)
         *
         */
        getDevStatus: async (deploymentId, namespace, name, options = {}) => {
            const localVarPath = `/v2/studio/deployment/dev-status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves the complete schema for a table including columns, indexes, and foreign keys.
         * @summary Get table schema
         * @param {string} tableName
         * @param {string} deploymentId Deployment ID (UUID)
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetTableSchema200Response>} A promise that resolves to a `GetTableSchema200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-table-schema)
         *
         */
        getTableSchema: async (tableName, deploymentId, fid, options = {}) => {
            // verify required parameter 'tableName' is not null or undefined
            (0, common_1.assertParamExists)('getTableSchema', 'tableName', tableName);
            // verify required parameter 'deploymentId' is not null or undefined
            (0, common_1.assertParamExists)('getTableSchema', 'deploymentId', deploymentId);
            const localVarPath = `/v2/studio/deployment/database/tables/{table_name}/schema`
                .replace(`{${"table_name"}}`, encodeURIComponent(String(tableName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Inserts one or more rows into the specified table. Returns the inserted rows with generated values.
         * @summary Insert rows into table
         * @param {string} tableName
         * @param {InsertRowsRequest} insertRowsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<InsertRows200Response>} A promise that resolves to a `InsertRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/insert-rows)
         *
         */
        insertRows: async (tableName, insertRowsRequest, options = {}) => {
            // verify required parameter 'tableName' is not null or undefined
            (0, common_1.assertParamExists)('insertRows', 'tableName', tableName);
            // verify required parameter 'insertRowsRequest' is not null or undefined
            (0, common_1.assertParamExists)('insertRows', 'insertRowsRequest', insertRowsRequest);
            const localVarPath = `/v2/studio/deployment/database/tables/{table_name}/rows`
                .replace(`{${"table_name"}}`, encodeURIComponent(String(tableName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(insertRowsRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all conversations for a specific deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List conversations for a deployment
         * @param {string} [deploymentId] Deployment ID (UUID). If provided, filters conversations to this deployment only.
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users. Studio admins can omit to query all conversations.
         * @param {string} [name] Kubernetes deployment name. If provided, filters conversations to this deployment only.
         * @param {boolean | null} [includeDeleted] Include deleted conversations in the response. Defaults to false.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListConversations200Response>} A promise that resolves to a `ListConversations200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-conversations)
         *
         */
        listConversations: async (deploymentId, fid, name, includeDeleted, options = {}) => {
            const localVarPath = `/v2/studio/deployment/conversations`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (includeDeleted !== undefined) {
                localVarQueryParameter['include_deleted'] = includeDeleted;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists files in a directory of the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List deployment files
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {string} [directory] Directory path relative to gen/ (defaults to root)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeploymentFiles200Response>} A promise that resolves to a `ListDeploymentFiles200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployment-files)
         *
         */
        listDeploymentFiles: async (deploymentId, fid, name, namespace, directory, options = {}) => {
            const localVarPath = `/v2/studio/deployment/files`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (directory !== undefined) {
                localVarQueryParameter['directory'] = directory;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all miniapp generator deployments for a user. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List deployments
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users. Studio admins can omit to query all deployments.
         * @param {number} [limit] Maximum number of deployments to return. Defaults to 50, max 1000.  (Default: 50, Maximum: 1000)
         * @param {number | null} [offset] Number of deployments to skip for pagination. Defaults to 0.
         * @param {boolean | null} [includeDeleted] Include deleted deployments in the response. Defaults to false.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<Array<ListDeployments200ResponseInner>>} A promise that resolves to a `Array<ListDeployments200ResponseInner>` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployments)
         *
         */
        listDeployments: async (fid, limit, offset, includeDeleted, options = {}) => {
            const localVarPath = `/v2/studio/deployment/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (offset !== undefined) {
                localVarQueryParameter['offset'] = offset;
            }
            if (includeDeleted !== undefined) {
                localVarQueryParameter['include_deleted'] = includeDeleted;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves all secrets for a deployment.
         * @summary List deployment secrets
         * @param {string} deploymentId Deployment ID to list secrets for
         * @param {string} [key] Optional filter by environment variable name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListSecrets200Response>} A promise that resolves to a `ListSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-secrets)
         *
         */
        listSecrets: async (deploymentId, key, options = {}) => {
            // verify required parameter 'deploymentId' is not null or undefined
            (0, common_1.assertParamExists)('listSecrets', 'deploymentId', deploymentId);
            const localVarPath = `/v2/studio/deployment/secrets/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (key !== undefined) {
                localVarQueryParameter['key'] = key;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all tables and views in the deployment database, excluding system tables.
         * @summary List all tables in deployment database
         * @param {string} deploymentId Deployment ID (UUID)
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListTables200Response>} A promise that resolves to a `ListTables200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-tables)
         *
         */
        listTables: async (deploymentId, fid, options = {}) => {
            // verify required parameter 'deploymentId' is not null or undefined
            (0, common_1.assertParamExists)('listTables', 'deploymentId', deploymentId);
            const localVarPath = `/v2/studio/deployment/database/tables`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Sends a prompt to a specific miniapp generator deployment and returns a streaming response using Server-Sent Events. The response is a continuous stream of Server-Sent Events, not a single JSON payload. Each event contains a JSON object with type, message, and other fields specific to the message type. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Prompt a deployment with streaming response
         * @param {PromptDeploymentStreamRequest} promptDeploymentStreamRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/prompt-deployment-stream)
         *
         */
        promptDeploymentStream: async (promptDeploymentStreamRequest, options = {}) => {
            // verify required parameter 'promptDeploymentStreamRequest' is not null or undefined
            (0, common_1.assertParamExists)('promptDeploymentStream', 'promptDeploymentStreamRequest', promptDeploymentStreamRequest);
            const localVarPath = `/v2/studio/deployment/prompt/stream`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(promptDeploymentStreamRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Query data from a table with pagination and sorting.
         * @summary Query table data
         * @param {QueryTableRequest} queryTableRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<QueryTable200Response>} A promise that resolves to a `QueryTable200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/query-table)
         *
         */
        queryTable: async (queryTableRequest, options = {}) => {
            // verify required parameter 'queryTableRequest' is not null or undefined
            (0, common_1.assertParamExists)('queryTable', 'queryTableRequest', queryTableRequest);
            const localVarPath = `/v2/studio/deployment/database/query`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(queryTableRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Attempts to recover a broken dev server. Phase 1: reads dev server error logs and sends them to an AI agent for fixing, then waits for HMR to auto-rebuild. Phase 2: if HMR fails, falls back to a full npm build with AI retry loop. Streams progress events via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Recover dev server with two-phase strategy
         * @param {RecoverRequest} recoverRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/recover)
         *
         */
        recover: async (recoverRequest, options = {}) => {
            // verify required parameter 'recoverRequest' is not null or undefined
            (0, common_1.assertParamExists)('recover', 'recoverRequest', recoverRequest);
            const localVarPath = `/v2/studio/deployment/recover`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(recoverRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Starts the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Start generated miniapp
         * @param {StartAppRequest} startAppRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/start-app)
         *
         */
        startApp: async (startAppRequest, options = {}) => {
            // verify required parameter 'startAppRequest' is not null or undefined
            (0, common_1.assertParamExists)('startApp', 'startAppRequest', startAppRequest);
            const localVarPath = `/v2/studio/deployment/start`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(startAppRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Stops the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Stop generated miniapp
         * @param {StartAppRequest} startAppRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/stop-app)
         *
         */
        stopApp: async (startAppRequest, options = {}) => {
            // verify required parameter 'startAppRequest' is not null or undefined
            (0, common_1.assertParamExists)('stopApp', 'startAppRequest', startAppRequest);
            const localVarPath = `/v2/studio/deployment/stop`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(startAppRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Updates rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk updates.
         * @summary Update rows in table
         * @param {string} tableName
         * @param {UpdateRowsRequest} updateRowsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpdateRows200Response>} A promise that resolves to a `UpdateRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/update-rows)
         *
         */
        updateRows: async (tableName, updateRowsRequest, options = {}) => {
            // verify required parameter 'tableName' is not null or undefined
            (0, common_1.assertParamExists)('updateRows', 'tableName', tableName);
            // verify required parameter 'updateRowsRequest' is not null or undefined
            (0, common_1.assertParamExists)('updateRows', 'updateRowsRequest', updateRowsRequest);
            const localVarPath = `/v2/studio/deployment/database/tables/{table_name}/rows`
                .replace(`{${"table_name"}}`, encodeURIComponent(String(tableName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PATCH' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateRowsRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Uploads an image file to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Upload image to deployment
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image)
         *
         */
        uploadImage: async (options = {}) => {
            const localVarPath = `/v2/studio/deployment/upload-image`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Downloads an image from the provided URL and saves it to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Upload image from URL to deployment
         * @param {UploadImageUrlRequest} uploadImageUrlRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image-url)
         *
         */
        uploadImageUrl: async (uploadImageUrlRequest, options = {}) => {
            // verify required parameter 'uploadImageUrlRequest' is not null or undefined
            (0, common_1.assertParamExists)('uploadImageUrl', 'uploadImageUrlRequest', uploadImageUrlRequest);
            const localVarPath = `/v2/studio/deployment/upload-image-url`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(uploadImageUrlRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Upsert secrets for a deployment.
         * @summary Upsert deployment secrets
         * @param {UpsertSecretsRequest} upsertSecretsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upsert-secrets)
         *
         */
        upsertSecrets: async (upsertSecretsRequest, options = {}) => {
            // verify required parameter 'upsertSecretsRequest' is not null or undefined
            (0, common_1.assertParamExists)('upsertSecrets', 'upsertSecretsRequest', upsertSecretsRequest);
            const localVarPath = `/v2/studio/deployment/secrets/`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(upsertSecretsRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves the build logs for a Vercel deployment. Useful for debugging failed deployments. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get Vercel deployment build logs
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user
         * @param {string} [namespace] K8s Namespace name
         * @param {string} [name] Deployment name used to identify the Vercel project. Required if deployment_id not provided.
         * @param {number} [limit] Maximum number of log events to return. Defaults to 100.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<VercelDeploymentLogs200Response>} A promise that resolves to a `VercelDeploymentLogs200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-logs)
         *
         */
        vercelDeploymentLogs: async (deploymentId, fid, namespace, name, limit, options = {}) => {
            const localVarPath = `/v2/studio/vercel/logs`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Retrieves the status of a Vercel deployment for a miniapp. Looks up the Vercel project ID from the deployment and queries Vercel API for deployment status. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get Vercel deployment status
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [namespace] K8s Namespace name
         * @param {string} [name] Deployment name used to identify the Vercel project. Required if deployment_id not provided.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<VercelDeploymentStatus200Response>} A promise that resolves to a `VercelDeploymentStatus200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-status)
         *
         */
        vercelDeploymentStatus: async (deploymentId, fid, namespace, name, options = {}) => {
            const localVarPath = `/v2/studio/vercel/status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication ApiKeyAuth required
            await (0, common_1.setApiKeyToObject)(localVarHeaderParameter, "x-api-key", configuration);
            if (deploymentId !== undefined) {
                localVarQueryParameter['deployment_id'] = deploymentId;
            }
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (namespace !== undefined) {
                localVarQueryParameter['namespace'] = namespace;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.StudioApiAxiosParamCreator = StudioApiAxiosParamCreator;
/**
 * StudioApi - functional programming interface
 * @export
 */
const StudioApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.StudioApiAxiosParamCreator)(configuration);
    return {
        /**
         * Associates a generated miniapp with a Farcaster account using a JFS-signed domain association. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Set account association
         * @param {AssociateDeploymentRequest} associateDeploymentRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<AssociateDeployment200Response>} A promise that resolves to a `AssociateDeployment200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/associate-deployment)
         *
         */
        async associateDeployment(associateDeploymentRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.associateDeployment(associateDeploymentRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.associateDeployment']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Runs Next.js build process for the generated app. If build fails, automatically calls a build-fixer agent to resolve errors. Streams build output and agent responses via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Build generated app with automatic error fixing
         * @param {BuildRequest} buildRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/build)
         *
         */
        async build(buildRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.build(buildRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.build']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Creates and deploys an instance of the miniapp generator for a user. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Create a miniapp generator deployment
         * @param {CreateDeploymentRequest} createDeploymentRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/create-deployment)
         *
         */
        async createDeployment(createDeploymentRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.createDeployment(createDeploymentRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.createDeployment']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Deletes a specific miniapp generator deployment or all deployments for a FID. If deployment_id or name is provided, deletes single deployment. If only FID is provided, deletes all deployments for that FID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Delete deployment(s)
         * @param {DeleteDeploymentRequest} deleteDeploymentRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeleteDeployment200Response>} A promise that resolves to a `DeleteDeployment200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-deployment)
         *
         */
        async deleteDeployment(deleteDeploymentRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteDeployment(deleteDeploymentRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.deleteDeployment']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Deletes rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk deletes.
         * @summary Delete rows from table
         * @param {string} tableName
         * @param {DeleteRowsRequest} deleteRowsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeleteRows200Response>} A promise that resolves to a `DeleteRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-rows)
         *
         */
        async deleteRows(tableName, deleteRowsRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteRows(tableName, deleteRowsRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.deleteRows']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Deletes environment variables (secrets) from a deployment.
         * @summary Delete deployment secrets
         * @param {DeleteSecretsRequest} deleteSecretsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-secrets)
         *
         */
        async deleteSecrets(deleteSecretsRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteSecrets(deleteSecretsRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.deleteSecrets']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Publishes the generated miniapp to Vercel via GitHub. Creates a GitHub repository, pushes code, creates a Vercel project linked to GitHub, and triggers deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Deploy miniapp to Vercel
         * @param {DeployToVercelRequest} deployToVercelRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeployToVercel200Response>} A promise that resolves to a `DeployToVercel200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/deploy-to-vercel)
         *
         */
        async deployToVercel(deployToVercelRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.deployToVercel(deployToVercelRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.deployToVercel']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Executes a raw SQL query against the deployment database. Only SELECT, WITH, and EXPLAIN queries are allowed. Admin access required.
         * @summary Execute raw SQL query (admin only)
         * @param {ExecuteSqlRequest} executeSqlRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ExecuteSql200Response>} A promise that resolves to a `ExecuteSql200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/execute-sql)
         *
         */
        async executeSql(executeSqlRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.executeSql(executeSqlRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.executeSql']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves the account-association.json file from a miniapp deployment, which contains the JFS-signed domain association. Requires API key authentication.
         * @summary Get account association of a miniapp
         * @param {string} [deploymentId] Deployment ID
         * @param {string} [namespace] Kubernetes namespace name
         * @param {string} [name] Kubernetes deployment name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetAccountAssociation200Response>} A promise that resolves to a `GetAccountAssociation200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-account-association)
         *
         */
        async getAccountAssociation(deploymentId, namespace, name, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.getAccountAssociation(deploymentId, namespace, name, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.getAccountAssociation']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves all messages in a specific conversation. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get messages in a conversation
         * @param {string} conversationId Conversation ID
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {boolean | null} [includeDeleted] Include deleted messages in the response. Defaults to false.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetConversationMessages200Response>} A promise that resolves to a `GetConversationMessages200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-conversation-messages)
         *
         */
        async getConversationMessages(conversationId, deploymentId, fid, name, namespace, includeDeleted, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.getConversationMessages(conversationId, deploymentId, fid, name, namespace, includeDeleted, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.getConversationMessages']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Fetches info about a miniapp generator deployment by its deployment_id or name and creator\'s Farcaster ID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get deployment info
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment)
         *
         */
        async getDeployment(deploymentId, fid, name, namespace, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDeployment(deploymentId, fid, name, namespace, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.getDeployment']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves the contents of a specific file from the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get deployment file contents
         * @param {string} filePath File path relative to gen/
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetDeploymentFile200Response>} A promise that resolves to a `GetDeploymentFile200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment-file)
         *
         */
        async getDeploymentFile(filePath, deploymentId, fid, name, namespace, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDeploymentFile(filePath, deploymentId, fid, name, namespace, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.getDeploymentFile']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves the dev-status.json file from a miniapp deployment, which tracks the progress of app development phases. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get dev status of a miniapp
         * @param {string} [deploymentId] Deployment ID
         * @param {string} [namespace] Kubernetes namespace name
         * @param {string} [name] Kubernetes deployment name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetDevStatus200Response>} A promise that resolves to a `GetDevStatus200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-dev-status)
         *
         */
        async getDevStatus(deploymentId, namespace, name, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDevStatus(deploymentId, namespace, name, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.getDevStatus']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves the complete schema for a table including columns, indexes, and foreign keys.
         * @summary Get table schema
         * @param {string} tableName
         * @param {string} deploymentId Deployment ID (UUID)
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetTableSchema200Response>} A promise that resolves to a `GetTableSchema200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-table-schema)
         *
         */
        async getTableSchema(tableName, deploymentId, fid, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.getTableSchema(tableName, deploymentId, fid, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.getTableSchema']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Inserts one or more rows into the specified table. Returns the inserted rows with generated values.
         * @summary Insert rows into table
         * @param {string} tableName
         * @param {InsertRowsRequest} insertRowsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<InsertRows200Response>} A promise that resolves to a `InsertRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/insert-rows)
         *
         */
        async insertRows(tableName, insertRowsRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.insertRows(tableName, insertRowsRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.insertRows']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Lists all conversations for a specific deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List conversations for a deployment
         * @param {string} [deploymentId] Deployment ID (UUID). If provided, filters conversations to this deployment only.
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users. Studio admins can omit to query all conversations.
         * @param {string} [name] Kubernetes deployment name. If provided, filters conversations to this deployment only.
         * @param {boolean | null} [includeDeleted] Include deleted conversations in the response. Defaults to false.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListConversations200Response>} A promise that resolves to a `ListConversations200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-conversations)
         *
         */
        async listConversations(deploymentId, fid, name, includeDeleted, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.listConversations(deploymentId, fid, name, includeDeleted, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.listConversations']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Lists files in a directory of the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List deployment files
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [name] Kubernetes deployment name. Required if deployment_id not provided.
         * @param {string} [namespace] Optional Kubernetes namespace. If not provided, will query for the active namespace for the given FID.
         * @param {string} [directory] Directory path relative to gen/ (defaults to root)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeploymentFiles200Response>} A promise that resolves to a `ListDeploymentFiles200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployment-files)
         *
         */
        async listDeploymentFiles(deploymentId, fid, name, namespace, directory, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.listDeploymentFiles(deploymentId, fid, name, namespace, directory, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.listDeploymentFiles']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Lists all miniapp generator deployments for a user. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List deployments
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users. Studio admins can omit to query all deployments.
         * @param {number} [limit] Maximum number of deployments to return. Defaults to 50, max 1000.  (Default: 50, Maximum: 1000)
         * @param {number | null} [offset] Number of deployments to skip for pagination. Defaults to 0.
         * @param {boolean | null} [includeDeleted] Include deleted deployments in the response. Defaults to false.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<Array<ListDeployments200ResponseInner>>} A promise that resolves to a `Array<ListDeployments200ResponseInner>` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployments)
         *
         */
        async listDeployments(fid, limit, offset, includeDeleted, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.listDeployments(fid, limit, offset, includeDeleted, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.listDeployments']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves all secrets for a deployment.
         * @summary List deployment secrets
         * @param {string} deploymentId Deployment ID to list secrets for
         * @param {string} [key] Optional filter by environment variable name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListSecrets200Response>} A promise that resolves to a `ListSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-secrets)
         *
         */
        async listSecrets(deploymentId, key, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.listSecrets(deploymentId, key, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.listSecrets']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Lists all tables and views in the deployment database, excluding system tables.
         * @summary List all tables in deployment database
         * @param {string} deploymentId Deployment ID (UUID)
         * @param {number | null} [fid] Farcaster ID of the user. Required for non-admin users.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListTables200Response>} A promise that resolves to a `ListTables200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-tables)
         *
         */
        async listTables(deploymentId, fid, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.listTables(deploymentId, fid, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.listTables']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Sends a prompt to a specific miniapp generator deployment and returns a streaming response using Server-Sent Events. The response is a continuous stream of Server-Sent Events, not a single JSON payload. Each event contains a JSON object with type, message, and other fields specific to the message type. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Prompt a deployment with streaming response
         * @param {PromptDeploymentStreamRequest} promptDeploymentStreamRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/prompt-deployment-stream)
         *
         */
        async promptDeploymentStream(promptDeploymentStreamRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.promptDeploymentStream(promptDeploymentStreamRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.promptDeploymentStream']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Query data from a table with pagination and sorting.
         * @summary Query table data
         * @param {QueryTableRequest} queryTableRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<QueryTable200Response>} A promise that resolves to a `QueryTable200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/query-table)
         *
         */
        async queryTable(queryTableRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.queryTable(queryTableRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.queryTable']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Attempts to recover a broken dev server. Phase 1: reads dev server error logs and sends them to an AI agent for fixing, then waits for HMR to auto-rebuild. Phase 2: if HMR fails, falls back to a full npm build with AI retry loop. Streams progress events via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Recover dev server with two-phase strategy
         * @param {RecoverRequest} recoverRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/recover)
         *
         */
        async recover(recoverRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.recover(recoverRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.recover']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Starts the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Start generated miniapp
         * @param {StartAppRequest} startAppRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/start-app)
         *
         */
        async startApp(startAppRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.startApp(startAppRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.startApp']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Stops the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Stop generated miniapp
         * @param {StartAppRequest} startAppRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/stop-app)
         *
         */
        async stopApp(startAppRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.stopApp(startAppRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.stopApp']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Updates rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk updates.
         * @summary Update rows in table
         * @param {string} tableName
         * @param {UpdateRowsRequest} updateRowsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpdateRows200Response>} A promise that resolves to a `UpdateRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/update-rows)
         *
         */
        async updateRows(tableName, updateRowsRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateRows(tableName, updateRowsRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.updateRows']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Uploads an image file to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Upload image to deployment
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image)
         *
         */
        async uploadImage(options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.uploadImage(options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.uploadImage']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Downloads an image from the provided URL and saves it to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Upload image from URL to deployment
         * @param {UploadImageUrlRequest} uploadImageUrlRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image-url)
         *
         */
        async uploadImageUrl(uploadImageUrlRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.uploadImageUrl(uploadImageUrlRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.uploadImageUrl']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Upsert secrets for a deployment.
         * @summary Upsert deployment secrets
         * @param {UpsertSecretsRequest} upsertSecretsRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upsert-secrets)
         *
         */
        async upsertSecrets(upsertSecretsRequest, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.upsertSecrets(upsertSecretsRequest, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.upsertSecrets']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves the build logs for a Vercel deployment. Useful for debugging failed deployments. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get Vercel deployment build logs
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user
         * @param {string} [namespace] K8s Namespace name
         * @param {string} [name] Deployment name used to identify the Vercel project. Required if deployment_id not provided.
         * @param {number} [limit] Maximum number of log events to return. Defaults to 100.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<VercelDeploymentLogs200Response>} A promise that resolves to a `VercelDeploymentLogs200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-logs)
         *
         */
        async vercelDeploymentLogs(deploymentId, fid, namespace, name, limit, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.vercelDeploymentLogs(deploymentId, fid, namespace, name, limit, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.vercelDeploymentLogs']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Retrieves the status of a Vercel deployment for a miniapp. Looks up the Vercel project ID from the deployment and queries Vercel API for deployment status. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get Vercel deployment status
         * @param {string} [deploymentId] Deployment ID (UUID). Required if name not provided.
         * @param {number | null} [fid] Farcaster ID of the user; if not provided, namespace must be provided
         * @param {string} [namespace] K8s Namespace name
         * @param {string} [name] Deployment name used to identify the Vercel project. Required if deployment_id not provided.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<VercelDeploymentStatus200Response>} A promise that resolves to a `VercelDeploymentStatus200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-status)
         *
         */
        async vercelDeploymentStatus(deploymentId, fid, namespace, name, options) {
            var _a, _b, _c;
            const localVarAxiosArgs = await localVarAxiosParamCreator.vercelDeploymentStatus(deploymentId, fid, namespace, name, options);
            const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
            const localVarOperationServerBasePath = (_c = (_b = base_1.operationServerMap['StudioApi.vercelDeploymentStatus']) === null || _b === void 0 ? void 0 : _b[localVarOperationServerIndex]) === null || _c === void 0 ? void 0 : _c.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.StudioApiFp = StudioApiFp;
/**
 * StudioApi - factory interface
 * @export
 */
const StudioApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.StudioApiFp)(configuration);
    return {
        /**
         * Associates a generated miniapp with a Farcaster account using a JFS-signed domain association. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Set account association
         * @param {StudioApiAssociateDeploymentRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<AssociateDeployment200Response>} A promise that resolves to a `AssociateDeployment200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/associate-deployment)
         *
         */
        associateDeployment(requestParameters, options) {
            return localVarFp.associateDeployment(requestParameters.associateDeploymentRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Runs Next.js build process for the generated app. If build fails, automatically calls a build-fixer agent to resolve errors. Streams build output and agent responses via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Build generated app with automatic error fixing
         * @param {StudioApiBuildRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/build)
         *
         */
        build(requestParameters, options) {
            return localVarFp.build(requestParameters.buildRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Creates and deploys an instance of the miniapp generator for a user. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Create a miniapp generator deployment
         * @param {StudioApiCreateDeploymentRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/create-deployment)
         *
         */
        createDeployment(requestParameters, options) {
            return localVarFp.createDeployment(requestParameters.createDeploymentRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Deletes a specific miniapp generator deployment or all deployments for a FID. If deployment_id or name is provided, deletes single deployment. If only FID is provided, deletes all deployments for that FID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Delete deployment(s)
         * @param {StudioApiDeleteDeploymentRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeleteDeployment200Response>} A promise that resolves to a `DeleteDeployment200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-deployment)
         *
         */
        deleteDeployment(requestParameters, options) {
            return localVarFp.deleteDeployment(requestParameters.deleteDeploymentRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Deletes rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk deletes.
         * @summary Delete rows from table
         * @param {StudioApiDeleteRowsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeleteRows200Response>} A promise that resolves to a `DeleteRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-rows)
         *
         */
        deleteRows(requestParameters, options) {
            return localVarFp.deleteRows(requestParameters.tableName, requestParameters.deleteRowsRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Deletes environment variables (secrets) from a deployment.
         * @summary Delete deployment secrets
         * @param {StudioApiDeleteSecretsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-secrets)
         *
         */
        deleteSecrets(requestParameters, options) {
            return localVarFp.deleteSecrets(requestParameters.deleteSecretsRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Publishes the generated miniapp to Vercel via GitHub. Creates a GitHub repository, pushes code, creates a Vercel project linked to GitHub, and triggers deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Deploy miniapp to Vercel
         * @param {StudioApiDeployToVercelRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<DeployToVercel200Response>} A promise that resolves to a `DeployToVercel200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/deploy-to-vercel)
         *
         */
        deployToVercel(requestParameters, options) {
            return localVarFp.deployToVercel(requestParameters.deployToVercelRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Executes a raw SQL query against the deployment database. Only SELECT, WITH, and EXPLAIN queries are allowed. Admin access required.
         * @summary Execute raw SQL query (admin only)
         * @param {StudioApiExecuteSqlRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ExecuteSql200Response>} A promise that resolves to a `ExecuteSql200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/execute-sql)
         *
         */
        executeSql(requestParameters, options) {
            return localVarFp.executeSql(requestParameters.executeSqlRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves the account-association.json file from a miniapp deployment, which contains the JFS-signed domain association. Requires API key authentication.
         * @summary Get account association of a miniapp
         * @param {StudioApiGetAccountAssociationRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetAccountAssociation200Response>} A promise that resolves to a `GetAccountAssociation200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-account-association)
         *
         */
        getAccountAssociation(requestParameters = {}, options) {
            return localVarFp.getAccountAssociation(requestParameters.deploymentId, requestParameters.namespace, requestParameters.name, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves all messages in a specific conversation. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get messages in a conversation
         * @param {StudioApiGetConversationMessagesRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetConversationMessages200Response>} A promise that resolves to a `GetConversationMessages200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-conversation-messages)
         *
         */
        getConversationMessages(requestParameters, options) {
            return localVarFp.getConversationMessages(requestParameters.conversationId, requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, requestParameters.includeDeleted, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetches info about a miniapp generator deployment by its deployment_id or name and creator\'s Farcaster ID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get deployment info
         * @param {StudioApiGetDeploymentRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment)
         *
         */
        getDeployment(requestParameters = {}, options) {
            return localVarFp.getDeployment(requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves the contents of a specific file from the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get deployment file contents
         * @param {StudioApiGetDeploymentFileRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetDeploymentFile200Response>} A promise that resolves to a `GetDeploymentFile200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment-file)
         *
         */
        getDeploymentFile(requestParameters, options) {
            return localVarFp.getDeploymentFile(requestParameters.filePath, requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves the dev-status.json file from a miniapp deployment, which tracks the progress of app development phases. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get dev status of a miniapp
         * @param {StudioApiGetDevStatusRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetDevStatus200Response>} A promise that resolves to a `GetDevStatus200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-dev-status)
         *
         */
        getDevStatus(requestParameters = {}, options) {
            return localVarFp.getDevStatus(requestParameters.deploymentId, requestParameters.namespace, requestParameters.name, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves the complete schema for a table including columns, indexes, and foreign keys.
         * @summary Get table schema
         * @param {StudioApiGetTableSchemaRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<GetTableSchema200Response>} A promise that resolves to a `GetTableSchema200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-table-schema)
         *
         */
        getTableSchema(requestParameters, options) {
            return localVarFp.getTableSchema(requestParameters.tableName, requestParameters.deploymentId, requestParameters.fid, options).then((request) => request(axios, basePath));
        },
        /**
         * Inserts one or more rows into the specified table. Returns the inserted rows with generated values.
         * @summary Insert rows into table
         * @param {StudioApiInsertRowsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<InsertRows200Response>} A promise that resolves to a `InsertRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/insert-rows)
         *
         */
        insertRows(requestParameters, options) {
            return localVarFp.insertRows(requestParameters.tableName, requestParameters.insertRowsRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Lists all conversations for a specific deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List conversations for a deployment
         * @param {StudioApiListConversationsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListConversations200Response>} A promise that resolves to a `ListConversations200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-conversations)
         *
         */
        listConversations(requestParameters = {}, options) {
            return localVarFp.listConversations(requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.includeDeleted, options).then((request) => request(axios, basePath));
        },
        /**
         * Lists files in a directory of the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List deployment files
         * @param {StudioApiListDeploymentFilesRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListDeploymentFiles200Response>} A promise that resolves to a `ListDeploymentFiles200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployment-files)
         *
         */
        listDeploymentFiles(requestParameters = {}, options) {
            return localVarFp.listDeploymentFiles(requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, requestParameters.directory, options).then((request) => request(axios, basePath));
        },
        /**
         * Lists all miniapp generator deployments for a user. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary List deployments
         * @param {StudioApiListDeploymentsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<Array<ListDeployments200ResponseInner>>} A promise that resolves to a `Array<ListDeployments200ResponseInner>` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployments)
         *
         */
        listDeployments(requestParameters = {}, options) {
            return localVarFp.listDeployments(requestParameters.fid, requestParameters.limit, requestParameters.offset, requestParameters.includeDeleted, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves all secrets for a deployment.
         * @summary List deployment secrets
         * @param {StudioApiListSecretsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListSecrets200Response>} A promise that resolves to a `ListSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-secrets)
         *
         */
        listSecrets(requestParameters, options) {
            return localVarFp.listSecrets(requestParameters.deploymentId, requestParameters.key, options).then((request) => request(axios, basePath));
        },
        /**
         * Lists all tables and views in the deployment database, excluding system tables.
         * @summary List all tables in deployment database
         * @param {StudioApiListTablesRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<ListTables200Response>} A promise that resolves to a `ListTables200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-tables)
         *
         */
        listTables(requestParameters, options) {
            return localVarFp.listTables(requestParameters.deploymentId, requestParameters.fid, options).then((request) => request(axios, basePath));
        },
        /**
         * Sends a prompt to a specific miniapp generator deployment and returns a streaming response using Server-Sent Events. The response is a continuous stream of Server-Sent Events, not a single JSON payload. Each event contains a JSON object with type, message, and other fields specific to the message type. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Prompt a deployment with streaming response
         * @param {StudioApiPromptDeploymentStreamRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/prompt-deployment-stream)
         *
         */
        promptDeploymentStream(requestParameters, options) {
            return localVarFp.promptDeploymentStream(requestParameters.promptDeploymentStreamRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Query data from a table with pagination and sorting.
         * @summary Query table data
         * @param {StudioApiQueryTableRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<QueryTable200Response>} A promise that resolves to a `QueryTable200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/query-table)
         *
         */
        queryTable(requestParameters, options) {
            return localVarFp.queryTable(requestParameters.queryTableRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Attempts to recover a broken dev server. Phase 1: reads dev server error logs and sends them to an AI agent for fixing, then waits for HMR to auto-rebuild. Phase 2: if HMR fails, falls back to a full npm build with AI retry loop. Streams progress events via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Recover dev server with two-phase strategy
         * @param {StudioApiRecoverRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<>} A promise that resolves to a `` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/recover)
         *
         */
        recover(requestParameters, options) {
            return localVarFp.recover(requestParameters.recoverRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Starts the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Start generated miniapp
         * @param {StudioApiStartAppRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/start-app)
         *
         */
        startApp(requestParameters, options) {
            return localVarFp.startApp(requestParameters.startAppRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Stops the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Stop generated miniapp
         * @param {StudioApiStopAppRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/stop-app)
         *
         */
        stopApp(requestParameters, options) {
            return localVarFp.stopApp(requestParameters.startAppRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Updates rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk updates.
         * @summary Update rows in table
         * @param {StudioApiUpdateRowsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpdateRows200Response>} A promise that resolves to a `UpdateRows200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/update-rows)
         *
         */
        updateRows(requestParameters, options) {
            return localVarFp.updateRows(requestParameters.tableName, requestParameters.updateRowsRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Uploads an image file to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Upload image to deployment
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image)
         *
         */
        uploadImage(options) {
            return localVarFp.uploadImage(options).then((request) => request(axios, basePath));
        },
        /**
         * Downloads an image from the provided URL and saves it to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Upload image from URL to deployment
         * @param {StudioApiUploadImageUrlRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image-url)
         *
         */
        uploadImageUrl(requestParameters, options) {
            return localVarFp.uploadImageUrl(requestParameters.uploadImageUrlRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Upsert secrets for a deployment.
         * @summary Upsert deployment secrets
         * @param {StudioApiUpsertSecretsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upsert-secrets)
         *
         */
        upsertSecrets(requestParameters, options) {
            return localVarFp.upsertSecrets(requestParameters.upsertSecretsRequest, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves the build logs for a Vercel deployment. Useful for debugging failed deployments. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get Vercel deployment build logs
         * @param {StudioApiVercelDeploymentLogsRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<VercelDeploymentLogs200Response>} A promise that resolves to a `VercelDeploymentLogs200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-logs)
         *
         */
        vercelDeploymentLogs(requestParameters = {}, options) {
            return localVarFp.vercelDeploymentLogs(requestParameters.deploymentId, requestParameters.fid, requestParameters.namespace, requestParameters.name, requestParameters.limit, options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieves the status of a Vercel deployment for a miniapp. Looks up the Vercel project ID from the deployment and queries Vercel API for deployment status. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
         * @summary Get Vercel deployment status
         * @param {StudioApiVercelDeploymentStatusRequest} requestParameters Request parameters.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         * @returns {Promise<VercelDeploymentStatus200Response>} A promise that resolves to a `VercelDeploymentStatus200Response` object
         *
         * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-status)
         *
         */
        vercelDeploymentStatus(requestParameters = {}, options) {
            return localVarFp.vercelDeploymentStatus(requestParameters.deploymentId, requestParameters.fid, requestParameters.namespace, requestParameters.name, options).then((request) => request(axios, basePath));
        },
    };
};
exports.StudioApiFactory = StudioApiFactory;
/**
 * StudioApi - object-oriented interface
 * @export
 * @class StudioApi
 * @extends {BaseAPI}
 */
class StudioApi extends base_1.BaseAPI {
    /**
     * Associates a generated miniapp with a Farcaster account using a JFS-signed domain association. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Set account association
     * @param {StudioApiAssociateDeploymentRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<AssociateDeployment200Response>} A promise that resolves to a `AssociateDeployment200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/associate-deployment)
     *
     */
    associateDeployment(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).associateDeployment(requestParameters.associateDeploymentRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Runs Next.js build process for the generated app. If build fails, automatically calls a build-fixer agent to resolve errors. Streams build output and agent responses via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Build generated app with automatic error fixing
     * @param {StudioApiBuildRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<>} A promise that resolves to a `` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/build)
     *
     */
    build(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).build(requestParameters.buildRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Creates and deploys an instance of the miniapp generator for a user. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Create a miniapp generator deployment
     * @param {StudioApiCreateDeploymentRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/create-deployment)
     *
     */
    createDeployment(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).createDeployment(requestParameters.createDeploymentRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes a specific miniapp generator deployment or all deployments for a FID. If deployment_id or name is provided, deletes single deployment. If only FID is provided, deletes all deployments for that FID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Delete deployment(s)
     * @param {StudioApiDeleteDeploymentRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<DeleteDeployment200Response>} A promise that resolves to a `DeleteDeployment200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-deployment)
     *
     */
    deleteDeployment(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).deleteDeployment(requestParameters.deleteDeploymentRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk deletes.
     * @summary Delete rows from table
     * @param {StudioApiDeleteRowsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<DeleteRows200Response>} A promise that resolves to a `DeleteRows200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-rows)
     *
     */
    deleteRows(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).deleteRows(requestParameters.tableName, requestParameters.deleteRowsRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes environment variables (secrets) from a deployment.
     * @summary Delete deployment secrets
     * @param {StudioApiDeleteSecretsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/delete-secrets)
     *
     */
    deleteSecrets(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).deleteSecrets(requestParameters.deleteSecretsRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Publishes the generated miniapp to Vercel via GitHub. Creates a GitHub repository, pushes code, creates a Vercel project linked to GitHub, and triggers deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Deploy miniapp to Vercel
     * @param {StudioApiDeployToVercelRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<DeployToVercel200Response>} A promise that resolves to a `DeployToVercel200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/deploy-to-vercel)
     *
     */
    deployToVercel(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).deployToVercel(requestParameters.deployToVercelRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Executes a raw SQL query against the deployment database. Only SELECT, WITH, and EXPLAIN queries are allowed. Admin access required.
     * @summary Execute raw SQL query (admin only)
     * @param {StudioApiExecuteSqlRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ExecuteSql200Response>} A promise that resolves to a `ExecuteSql200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/execute-sql)
     *
     */
    executeSql(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).executeSql(requestParameters.executeSqlRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves the account-association.json file from a miniapp deployment, which contains the JFS-signed domain association. Requires API key authentication.
     * @summary Get account association of a miniapp
     * @param {StudioApiGetAccountAssociationRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<GetAccountAssociation200Response>} A promise that resolves to a `GetAccountAssociation200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-account-association)
     *
     */
    getAccountAssociation(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).getAccountAssociation(requestParameters.deploymentId, requestParameters.namespace, requestParameters.name, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves all messages in a specific conversation. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Get messages in a conversation
     * @param {StudioApiGetConversationMessagesRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<GetConversationMessages200Response>} A promise that resolves to a `GetConversationMessages200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-conversation-messages)
     *
     */
    getConversationMessages(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).getConversationMessages(requestParameters.conversationId, requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, requestParameters.includeDeleted, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetches info about a miniapp generator deployment by its deployment_id or name and creator\'s Farcaster ID. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Get deployment info
     * @param {StudioApiGetDeploymentRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ListDeployments200ResponseInner>} A promise that resolves to a `ListDeployments200ResponseInner` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment)
     *
     */
    getDeployment(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).getDeployment(requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves the contents of a specific file from the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Get deployment file contents
     * @param {StudioApiGetDeploymentFileRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<GetDeploymentFile200Response>} A promise that resolves to a `GetDeploymentFile200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-deployment-file)
     *
     */
    getDeploymentFile(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).getDeploymentFile(requestParameters.filePath, requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves the dev-status.json file from a miniapp deployment, which tracks the progress of app development phases. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Get dev status of a miniapp
     * @param {StudioApiGetDevStatusRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<GetDevStatus200Response>} A promise that resolves to a `GetDevStatus200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-dev-status)
     *
     */
    getDevStatus(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).getDevStatus(requestParameters.deploymentId, requestParameters.namespace, requestParameters.name, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves the complete schema for a table including columns, indexes, and foreign keys.
     * @summary Get table schema
     * @param {StudioApiGetTableSchemaRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<GetTableSchema200Response>} A promise that resolves to a `GetTableSchema200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/get-table-schema)
     *
     */
    getTableSchema(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).getTableSchema(requestParameters.tableName, requestParameters.deploymentId, requestParameters.fid, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Inserts one or more rows into the specified table. Returns the inserted rows with generated values.
     * @summary Insert rows into table
     * @param {StudioApiInsertRowsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<InsertRows200Response>} A promise that resolves to a `InsertRows200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/insert-rows)
     *
     */
    insertRows(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).insertRows(requestParameters.tableName, requestParameters.insertRowsRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all conversations for a specific deployment. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary List conversations for a deployment
     * @param {StudioApiListConversationsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ListConversations200Response>} A promise that resolves to a `ListConversations200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-conversations)
     *
     */
    listConversations(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).listConversations(requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.includeDeleted, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists files in a directory of the generated app. Requires Studio admin authentication or FID ownership validation. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary List deployment files
     * @param {StudioApiListDeploymentFilesRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ListDeploymentFiles200Response>} A promise that resolves to a `ListDeploymentFiles200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployment-files)
     *
     */
    listDeploymentFiles(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).listDeploymentFiles(requestParameters.deploymentId, requestParameters.fid, requestParameters.name, requestParameters.namespace, requestParameters.directory, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all miniapp generator deployments for a user. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary List deployments
     * @param {StudioApiListDeploymentsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<Array<ListDeployments200ResponseInner>>} A promise that resolves to a `Array<ListDeployments200ResponseInner>` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-deployments)
     *
     */
    listDeployments(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).listDeployments(requestParameters.fid, requestParameters.limit, requestParameters.offset, requestParameters.includeDeleted, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves all secrets for a deployment.
     * @summary List deployment secrets
     * @param {StudioApiListSecretsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ListSecrets200Response>} A promise that resolves to a `ListSecrets200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-secrets)
     *
     */
    listSecrets(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).listSecrets(requestParameters.deploymentId, requestParameters.key, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all tables and views in the deployment database, excluding system tables.
     * @summary List all tables in deployment database
     * @param {StudioApiListTablesRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<ListTables200Response>} A promise that resolves to a `ListTables200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/list-tables)
     *
     */
    listTables(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).listTables(requestParameters.deploymentId, requestParameters.fid, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Sends a prompt to a specific miniapp generator deployment and returns a streaming response using Server-Sent Events. The response is a continuous stream of Server-Sent Events, not a single JSON payload. Each event contains a JSON object with type, message, and other fields specific to the message type. Requires authentication via API key in the request header. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Prompt a deployment with streaming response
     * @param {StudioApiPromptDeploymentStreamRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<>} A promise that resolves to a `` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/prompt-deployment-stream)
     *
     */
    promptDeploymentStream(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).promptDeploymentStream(requestParameters.promptDeploymentStreamRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Query data from a table with pagination and sorting.
     * @summary Query table data
     * @param {StudioApiQueryTableRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<QueryTable200Response>} A promise that resolves to a `QueryTable200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/query-table)
     *
     */
    queryTable(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).queryTable(requestParameters.queryTableRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Attempts to recover a broken dev server. Phase 1: reads dev server error logs and sends them to an AI agent for fixing, then waits for HMR to auto-rebuild. Phase 2: if HMR fails, falls back to a full npm build with AI retry loop. Streams progress events via Server-Sent Events. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Recover dev server with two-phase strategy
     * @param {StudioApiRecoverRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<>} A promise that resolves to a `` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/recover)
     *
     */
    recover(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).recover(requestParameters.recoverRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Starts the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Start generated miniapp
     * @param {StudioApiStartAppRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/start-app)
     *
     */
    startApp(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).startApp(requestParameters.startAppRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Stops the Next.js development server for the generated miniapp. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Stop generated miniapp
     * @param {StudioApiStopAppRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<StartApp200Response>} A promise that resolves to a `StartApp200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/stop-app)
     *
     */
    stopApp(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).stopApp(requestParameters.startAppRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates rows matching the WHERE conditions. WHERE clause is required to prevent accidental bulk updates.
     * @summary Update rows in table
     * @param {StudioApiUpdateRowsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<UpdateRows200Response>} A promise that resolves to a `UpdateRows200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/update-rows)
     *
     */
    updateRows(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).updateRows(requestParameters.tableName, requestParameters.updateRowsRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Uploads an image file to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Upload image to deployment
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image)
     *
     */
    uploadImage(options) {
        return (0, exports.StudioApiFp)(this.configuration).uploadImage(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Downloads an image from the provided URL and saves it to the generated miniapp public folder. The image will be accessible as a static asset on the deployed miniapp. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Upload image from URL to deployment
     * @param {StudioApiUploadImageUrlRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<UploadImage200Response>} A promise that resolves to a `UploadImage200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upload-image-url)
     *
     */
    uploadImageUrl(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).uploadImageUrl(requestParameters.uploadImageUrlRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Upsert secrets for a deployment.
     * @summary Upsert deployment secrets
     * @param {StudioApiUpsertSecretsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<UpsertSecrets200Response>} A promise that resolves to a `UpsertSecrets200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/upsert-secrets)
     *
     */
    upsertSecrets(requestParameters, options) {
        return (0, exports.StudioApiFp)(this.configuration).upsertSecrets(requestParameters.upsertSecretsRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves the build logs for a Vercel deployment. Useful for debugging failed deployments. Requires Studio admin authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Get Vercel deployment build logs
     * @param {StudioApiVercelDeploymentLogsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<VercelDeploymentLogs200Response>} A promise that resolves to a `VercelDeploymentLogs200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-logs)
     *
     */
    vercelDeploymentLogs(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).vercelDeploymentLogs(requestParameters.deploymentId, requestParameters.fid, requestParameters.namespace, requestParameters.name, requestParameters.limit, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieves the status of a Vercel deployment for a miniapp. Looks up the Vercel project ID from the deployment and queries Vercel API for deployment status. Requires API key authentication. Note: Studio CU is tracked based on LLM token usage, not per API call.
     * @summary Get Vercel deployment status
     * @param {StudioApiVercelDeploymentStatusRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StudioApi
     * @returns {Promise<VercelDeploymentStatus200Response>} A promise that resolves to a `VercelDeploymentStatus200Response` object
     *
     * For more information, refer to the [API documentation](https://docs.neynar.com/reference/vercel-deployment-status)
     *
     */
    vercelDeploymentStatus(requestParameters = {}, options) {
        return (0, exports.StudioApiFp)(this.configuration).vercelDeploymentStatus(requestParameters.deploymentId, requestParameters.fid, requestParameters.namespace, requestParameters.name, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.StudioApi = StudioApi;
